x-pack/plugins/ml/public/application/routing/use_resolver.ts

```diff
@@ -52,36 +53,52 @@ export const useResolver = (
         return;
       }
 
-      if (indexPatternId !== undefined || savedSearchId !== undefined) {
-        try {
-          // note, currently we're using our own kibana context that requires a current index pattern to be set
-          // this means, if the page uses this context, useResolver must be passed a string for the index pattern id
-          // and loadIndexPatterns must be part of the resolvers.
-          const { indexPattern, savedSearch } =
-            savedSearchId !== undefined
-              ? await getIndexPatternAndSavedSearch(savedSearchId)
-              : { savedSearch: null, indexPattern: await getIndexPatternById(indexPatternId!) };
+      try {
+        if (indexPatternId === '') {
+          throw new Error(
+            i18n.translate('xpack.ml.useResolver.errorIndexPatternIdEmptyString', {
+              defaultMessage: 'indexPatternId must not be empty string.',
+            })
+          );
+        }
 
-          const { combinedQuery } = createSearchItems(config, indexPattern!, savedSearch);
+        // note, currently we're using our own kibana context that requires a current index pattern to be set
```

Comment:
Is it worth adding a comment to clarify that for pages that don't require an index pattern, `undefined` should be passed for the index pattern and saved search IDs?

Reply:
Added comments here: 2a916a9